Phoenix/Core/Source/Voxels/Blocks.cpp

```diff
@@ -38,28 +36,26 @@ RegisteredBlock::RegisteredBlock(const std::string& unique, int id,
 
 RegisteredBlock::~RegisteredBlock() = default;
 
-BlockRegistry::BlockRegistry() : i(0) {};
-
-BlockRegistry::~BlockRegistry() {};
+BlockRegistry::BlockRegistry() : m_BlockIdCounter(0) {};
 
 int BlockRegistry::registerBlock(const std::string& uniqueName,
                                  const std::string& displayName)
 {
-	i++;
-	Blocks[i] = RegisteredBlock(uniqueName, i, displayName);
-	return i;
+	m_BlockIdCounter++;
+	m_Blocks.push_back(RegisteredBlock(uniqueName, m_BlockIdCounter, displayName));
+	return m_BlockIdCounter;
 };
 
 const std::string& BlockRegistry::getDisplayName(int blockId)
 {
-	return Blocks[blockId].displayName;
```

Comment:
Maybe assert or do some form of error checking here that `blockId` is a valid index.
Also does this work - because the ID's appear to be 1 based - since you do
```
m_blockIdCounter++;
m_blocks.push_back(RegisteredBlock(..., m_blockIdCounter, ...));
```
meaning that the first block in the vector (with index 0), will have an ID of 1 (since you incrememted m_blockIdCounter before pushing back).

So here you use blockId as an index straight into the vector, a block ID of 1 (the first registered block) will get the 2nd item in the array.

If that makes any sense 😄 

Also this using the block ID straight as an index doesn't take into account that blocks may be "unregistered"/removed from the blocks array.

Reply:
I fixed the 0 based issue, we should never remove a block from the array (Unless I missed something, this shouldn't be possible)